src/Filters/stream_filt.jl

```diff
@@ -565,18 +557,18 @@ end
 # Arbitrary resampling
 #
 # Updates FIRArbitrary state. See Section 7.5.1 in [1].
-#   [1] uses a phase accumilator that increments by Δ (Nϕ/rate)
+#   [1] uses a phase accumulator that increments by Δ (Nϕ/rate)
 
 function update!(kernel::FIRArbitrary)
     kernel.ϕAccumulator += kernel.Δ
 
-    if kernel.ϕAccumulator > kernel.Nϕ
-        kernel.xIdx        += div(kernel.ϕAccumulator-1, kernel.Nϕ)
-        kernel.ϕAccumulator = mod(kernel.ϕAccumulator-1, kernel.Nϕ) + 1
+    if kernel.ϕAccumulator >= kernel.Nϕ
+        Δx, kernel.ϕAccumulator = divrem(kernel.ϕAccumulator, kernel.Nϕ)
+        kernel.xIdx += round(Int, Δx)
```

Comment:
```suggestion
        kernel.xIdx += Int(Δx)
```
just to make it explicit that this should not actually round, but just be a type conversion?